src/lib/network_pool/intf.ml

```diff
@@ -133,3 +134,39 @@ module type Snark_pool_diff_intf = sig
 
   val apply : resource_pool -> t Envelope.Incoming.t -> t Deferred.Or_error.t
 end
+
+module type Transaction_pool_diff_intf = sig
+  module Stable : sig
+    module V1 : sig
+      type t = User_command.Stable.V1.t list
+      [@@deriving sexp, to_yojson, bin_io, version]
+    end
+
+    module Latest = V1
+  end
+
+  type t = Stable.Latest.t [@@deriving sexp, to_yojson]
+end
+
+module type Transaction_resource_pool_intf = sig
+  type t
+
+  type best_tip_diff
+
+  type transition_frontier
+
+  include
+    Resource_pool_base_intf
+    with type transition_frontier := transition_frontier
+     and type t := t
+
+  val pool : t -> Indexed_pool.t
```

Comment:
`Indexed_pool` is a purely functional data structure. Exposing `Indexed_pool` would not have many adverse effects.

Anyways, since the only function that uses the type `Indexed_pool` is `mem` for my PR, I am just going to expose `mem` instead